src/blocking/spi.rs

```diff
@@ -108,3 +108,134 @@ pub mod write_iter {
         }
     }
 }
+
+/// Provides SpiWithCS wrapper around an spi::* and OutputPin impl
+pub mod spi_with_cs {
+
+    use core::marker::PhantomData;
+    use core::fmt::Debug;
+
+    use crate::digital::OutputPin;
+    use super::*;
+
+    /// SpiWithCS wraps an blocking::spi* implementation with Chip Select (CS)
+    /// pin management.
+    /// For sharing SPI between peripherals, see [shared-bus]()
+    pub struct SpiWithCs<Spi, SpiError, Pin, PinError> {
+        spi: Spi,
+        cs: Pin,
+
+        _spi_err: PhantomData<SpiError>,
+        _pin_err: PhantomData<PinError>,
```

Comment:
Is this necessary?

Reply:
the associated errors? not _strictly_ but ime it makes the use of the objects a bit clearer having the types consistently bound to the object rather than able to vary by impl block.